$OpenBSD: patch-configure,v 1.7 2013/01/23 18:18:41 jeremy Exp $

Fix so name.

Override the arch setting to remove OpenBSD version from it,
so ports don't have to be bumped when OpenBSD version changes.

--- configure.orig	Wed Jan 16 20:57:31 2013
+++ configure	Thu Jan 17 08:59:36 2013
@@ -39481,7 +39481,7 @@ _ACEOF
 	 ;;
   openbsd*)
 	SOLIBS='$(LIBS)'
-	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.$(MAJOR).'`expr ${MINOR} \* 10 + ${TEENY}`
+	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.'${LIBruby19_VERSION}
 	 ;;
   solaris*)
 	SOLIBS='$(LIBS)'
@@ -40500,7 +40500,10 @@ _ACEOF
 _ACEOF
 
 else
-    arch="${target_cpu}-${target_os}"
+    case "$target_os" in
+      openbsd*) arch="$target_cpu-openbsd" ;;
+      *) arch="${target_cpu}-${target_os}" ;;
+    esac
     cat >>confdefs.h <<_ACEOF
 #define RUBY_PLATFORM "${arch}"
 _ACEOF
